### PR TITLE
Add a retry on exception to scroll click method

### PR DESCRIPTION
**Pull request by LandoCalrizzian**
During a scroll->click action a WebDriverException will be registered
because the element to click is beneath another element (typically the
cancel-back-next bar).
This changed adds addtional (default:1) attempts to scroll then click on
the element when a WebDriverException occurs

### CODE REVIEW

File: pages/page.py [hunk 1]
```diff
@@ -98,15 +98,29 @@ def ajax_complete(self, driver):
...
+                self.scroll_to_element(element)
+
+            try:
+                element.click()
+                break
+            except WebDriverException:
+                # Raise the click exception if scrolling disabled or
+                # max number of scroll->click attempts
+                if not scroll or attempts >= click_attempts_max:
+                    raise
```
Comment (jamesooden) [inline pages/page.py]: Only thing I might do is create a custom exception so you say how many attempts were made, and maybe if there is some information on the element put that too.

File: pages/page.py [hunk 2]
```diff
@@ -98,15 +98,29 @@ def ajax_complete(self, driver):
...
+         scroll the element into view then click on it.
 
-        element.click()
+        If the click attempt fails with WebDriverException then retry
+         scroll/click until the total number of attempts equals the max number
+         of attempts
+        """
+        attempts = 0
+        while attempts < click_attempts_max:
+            attempts += 1
```
Comment (apagac) [inline pages/page.py]: This seems to me more like a C code than python. What about using:

`for attempts in range(click_attempts_max):` 

This way, you won't have to explicitly zero or increment the "attempts" variable.

**Comment by apagac** [comment]
👍

File: pages/page.py [hunk 3]
```diff
@@ -107,21 +107,24 @@ def click(self, element, scroll=True, click_attempts_max=2):
          scroll/click until the total number of attempts equals the max number
          of attempts
         """
-        attempts = 0
-        while attempts < click_attempts_max:
-            attempts += 1
+        for attempts in xrange(0, click_attempts_max):
```
Comment (jamesooden) [inline pages/page.py]: Why this change?

Comment (jamesooden) [inline pages/page.py]: I understand, never mind
